examples/ecs/state.rs

```diff
@@ -147,9 +147,9 @@ fn change_color(
 ) {
     for handle in query.iter() {
         let material = assets.get_mut(handle).unwrap();
-        material
-            .color
-            .set_b((time.seconds_since_startup() * 5.0).sin() as f32 + 2.0);
+        if let Color::Rgba { blue, .. } = &mut material.color {
```

Comment:
I think this is a pretty major downside to the current approach. Interacting with srgb `r`, `g`, and `b` components directly will almost certainly be high-frequency operations.  `r()` and `set_r()` ops will be missed. 99% of people shouldn't even need to know that Color could have other representations.

Reply:
This could be solved by doing internal conversions as necessary to get srgb components, or by breaking out each Color enum variant into its own struct (and doing conversions a necessary). That would mean that Color would need to be backed by srgb (and we would have HslColor, LinearSrgbColor, etc and From impls for conversions).

I _think_ I like the enum (with internal conversions) more. But I'm curious what you think.

Reply:
I mainly like the enum more because it protects against conversion error unless absolutely necessary. But it also complicates the api / forces us to make hard choices about implicit behaviors.

Reply:
I'm not sure setting only a specific channel of the color is something that happens often in a game. I don't have an experience with "real" game development, but when I tried I handpicked a few colors, set them to consts with explicit names and swapped the colors completely

Reply:
Interacting with individual color channels comes into play for things like "animated materials", where you might want something to get more/less red over some period of time. Animating color happens in most visually polished games (although sometimes it happens inside shaders and often its an interpolation between two pre-defined colors). Ex: explosions, shield effects, outlines, ui transitions, etc. Animating channels individually is not uncommon.

```rust
if let Color::Rgba { red, .. } = &mut material.color {
  *red = time.sin() * speed
} else {
   // by not handling other cases we are introducing silent error cases
   // if we handle each type individually or add error cases this gets very boilerplate-ey
}

// easier to type, easier to read, no need to handle each error case (but also implicitly does lossy conversions if not stored as rgba)
material.color.set_r(time.sin() * speed)
```

All major engines (Unreal, Unity, Godot) have color channel properties/fields. I think our users will expect similar apis.

Reply:
I think whether or not Color should be an enum (or a simple sRGB struct) is a longer conversation. By adding the getters/setters we also buy time for that conversation by making this a non-breaking change.

Reply:
(because we do need the color fixes in this pr for 0.5 :smile:)

Reply:
fair enough, I can add setter/getter for rgb that will convert the color to rgba if needed 👍

Reply:
Awesome. Much appreciated. There are a lot of tradeoffs to discuss and its very possible that we'll end up with the exact impl currently in this pr (by removing the getters / setters), but I don't want to rush that decision now.